tools/run_tests/run_xds_tests.py

```diff
@@ -3321,6 +3411,9 @@ def __init__(self, compute, alpha_compute, project, project_num):
             logger.error('Test case(s) %s failed', failed_tests)
             sys.exit(1)
 finally:
-    if not args.keep_gcp_resources:
+    keep_resources = args.keep_gcp_resources
+    if args.halt_after_fail and failed_tests:
+        keep_resources = True
```

Comment:
Maybe put something in the log here saying that we intentionally are skipping clean up due to halt_after_fail?

Reply:
Done. Added an info log.